test/integration/029_docs_generate_tests/test_docs_generate.py

```diff
@@ -1884,10 +1868,10 @@ def expected_run_results(self, quote_schema=True, quote_model=False,
                     },
                     'description': 'The test model',
                     'docrefs': [],
-                    'empty': False,
                     'extra_ctes': [],
                     'extra_ctes_injected': True,
                     'fqn': ['test', 'model'],
+                    'index': None,
```

Comment:
In `run_results` I did `omit_none=False`, because a number of top-level fields that really seemed important (such as `error` and `fail`) can be `None`. I don't think this was a very satisfying solution, maybe the answer here is to override to_dict with special handling for those keys, and always set `omit_none=True`

Reply:
yeah, i think that could be a good answer! Good call to note that `error` and `fail` are significant to include here

Reply:
Ok. Let me know if you want me to do this as part of this PR, or a future one. I like the idea of tagging certain fields as "never omit" using field metadata, which seems like it'd be a change to hologram and another PR.